raccoochat/RaccooChatServer.cpp

```diff
@@ -69,65 +71,62 @@ bool RaccooChatHandler::validateName(const std::string& userName) {
 }
 
 bool RaccooChatHandler::checkPassword(const std::string& userName, const std::string& userPassword) {
-  if (usersOnline_[userName] == codePassword(userPassword)) return true;
+  if (registeredUsers_[userName].userPassword == codePassword(userPassword)) return true;
   else return false;
 }
 
 bool RaccooChatHandler::connectUser(const std::string& userName, const std::string& userPassword) {
-  usersData_[userName] = createNewUser();
-  usersOnline_[userName] = codePassword(userPassword);
+  registeredUsers_[userName] = createNewUser(userPassword);
+  const auto& currentUserIndex = registeredUsers_[userName].userIndex;
+  usersData_[currentUserIndex] = createNewUserData(currentUserIndex, userName);
   LOG(INFO) << "User " << userName << "'s connected to the chat";
   return true;
 }
 
 void RaccooChatHandler::disconnectUser(const std::string& userName) {
-  const auto& currentUserId = usersData_[userName].userId;
-  usersData_.erase(userName);
-  privateMessages_.erase(currentUserId);
+  const auto& currentUserIndex = registeredUsers_[userName].userIndex;
+  usersData_[currentUserIndex].status = 0;
   LOG(INFO) << "User " << userName << "'s left from the chat";
 }
 
-void RaccooChatHandler::getAllOnlineUsers(std::map<std::string, std::string>& _return) {
-  _return = usersOnline_;
+void RaccooChatHandler::getAllOnlineUsers(std::vector<std::string>& _return) {
+  std::vector<std::string> usersOnline;
+  for (auto& user : usersData_) {
+    if (user.second.status == 1) {
+      usersOnline.push_back(user.second.userName);
+    }
+  }
+  _return = usersOnline;
```

Comment:
you should use std::move here, to avoid extra copy, i.e.
_return = std::move(usersOnline)

or alternatively just work with _return all the time, without creating usersOnline